src/Illuminate/Mail/Mailable.php

```diff
@@ -438,6 +453,20 @@ protected function buildAttachments($message)
             );
         }
 
+        foreach ($this->embeds as $embed) {
+            $message->embed(Attachment::fromPath($embed['file'])
+                ->as($embed['options']['as'])
+                ->withMime($embed['options']['mime'] ?? null));
+        }
```

Comment:
You will notice this re-creates the `Attachment` class. This is to work around the inconsistent `Mail\Message` embed API.